File: frontend/src/services/templateService.ts
import api from "./api";

import type {
  MessageTemplate,
  MessageTemplateRequest,
} from "../types/template";

export const getTemplates = async (): Promise<MessageTemplate[]> => {
  const res = await api.get<MessageTemplate[]>("/templates/");
  return res.data;
};

export const createTemplate = async (
  data: MessageTemplateRequest
): Promise<MessageTemplate> => {
  const res = await api.post<MessageTemplate>("/templates/", data);
  return res.data;
};

export const updateTemplate = async (
  id: number,
  data: MessageTemplateRequest
): Promise<MessageTemplate> => {
  const res = await api.put<MessageTemplate>(`/templates/${id}/`, data);
  return res.data;
};

export const deleteTemplate = async (id: number): Promise<void> => {
  await api.delete(`/templates/${id}/`);
};

File: backend/app/services/template_service.py
from sqlalchemy.orm import Session
from app.models.template import MessageTemplate
from app.schemas.template import TemplateCreate


def create_template(db: Session, template: TemplateCreate):
    obj = MessageTemplate(**template.model_dump())
    db.add(obj)
    db.commit()
    db.refresh(obj)
    return obj


def get_templates(db: Session):
    return db.query(MessageTemplate).all()


def update_template(db: Session, template_id: int, template: TemplateCreate):
    obj = db.query(MessageTemplate).filter(
        MessageTemplate.id == template_id
    ).first()

    if not obj:
        return None

    obj.template_name = template.template_name
    obj.delivery_channel = template.delivery_channel
    obj.message_configuration = template.message_configuration

    db.commit()
    db.refresh(obj)

    return obj


def delete_template(db: Session, template_id: int):
    obj = db.query(MessageTemplate).filter(
        MessageTemplate.id == template_id
    ).first()

    if not obj:
        return None

    db.delete(obj)
    db.commit()

    return {"message": "Template deleted successfully"}